src/services/HitTest.ts

```diff
@@ -50,64 +72,111 @@ export class HitTest extends Emitter {
     );
   }
 
-  public load(items: HitBox[]) {
-    this.tree.load(items);
+  /**
+   * Load array of HitBox items
+   * @param items Array of HitBox items to load
+   * @returns void
+   */
+  public load(items: HitBox[]): void {
+    // For usableRectTracker use incremental addition (optimal)
+    this.usableRectTracker.clear();
+    for (const item of items) {
+      if (item.affectsUsableRect) {
+        this.usableRectTracker.add(item);
+      }
+    }
+
+    this.interactiveTree.load(items);
   }
 
-  // Single debounced job replaces complex scheduler logic
+  // Optimized debounced processing with incremental updates for drag operations
   protected processQueue = debounce(
     () => {
-      const items = [];
+      const interactiveItems = [];
+
```

Comment:
You should here check like:
```tsx
let shouldUpdateUsableRect = false
```
and here on 102 line:
```tsx
if (item.affectsUsableRect) {
   shouldUpdateUsableRect = true
   ...
```
this is because this.updateUsableRect always returns a new object, which causes the positions of the blocks whose positions depend on usabelRect to change.

Or deepEqual updateUsableRect value always before update it